src/snmp.c

```diff
@@ -1683,6 +1699,13 @@ static int csnmp_read_table(host_definition_t *host, data_definition_t *data) {
       break;
     }
 
+    if (req->command == SNMP_MSG_GETBULK) {
+      /* In bulk mode the host will send 'max_repetitions' values per
+         requested variable, so we need to split it per number of variable
+         to stay 'in budget' */
+      req->max_repetitions = floor(host->bulk_size / oid_list_todo_num);
```

Comment:
...and then is overridden by setting max_repetitions a second time here.

The formula floor(host->bulk_size / oid_list_todo_num) looks like it doesn't make sense. If oid_list_todo_num is greater than bulk_size, the whole equation evaluates to zero, which as I understand means "disabled".

This code as it stands make it impossible to set max_repetitions to one, as recommended here: https://rt.cpan.org/Public/Ticket/Attachment/WithHeaders/385924